velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -316,4 +316,41 @@ struct ArraySumFunction {
   }
 };
 
+template <typename T>
+struct ArrayAverageFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T)
+  template <typename TOutput, typename TInput>
+  FOLLY_ALWAYS_INLINE bool call(TOutput& out, const TInput& array) {
+    TOutput sum = 0;
+    int count = 0;
+    for (const auto& item : array) {
+      if (item.has_value()) {
+        sum += *item;
+        count++;
+      }
+    }
+    if (count != 0) {
+      out = sum / count;
+    }
+    return count != 0;
+  }
+
+  template <typename TOutput, typename TInput>
+  FOLLY_ALWAYS_INLINE void callNullFree(TOutput& out, const TInput& array) {
```

Comment:
To be consistent with the implementation of "call" and the Presto documentation this should return null if the input array is empty, so the return type should be bool.

Reply:
Change made.